logbook-servlet/src/main/java/org/zalando/logbook/servlet/TeeResponse.java

```diff
@@ -71,14 +70,20 @@ public Origin getOrigin() {
     }
 
     @Override
-    public Multimap<String, String> getHeaders() {
-        final Multimap<String, String> headers = ArrayListMultimap.create();
+    public ListMultimap<String, String> getHeaders() {
+        this.headers = Optional.ofNullable(headers)
```

Comment:
> I'd prefer an if statement here instead of constantly re-assigning the field. That would also tell you if the case is actually covered by tests :)

You have eyes like a hawk. To be honest that was the exact reason to do it like that, to trick code coverage m)